lib/puppet/transaction/report.rb

```diff
@@ -282,6 +289,8 @@ def initialize_from_hash(data)
       end
       @resource_statuses[record[0]] = status
     end
+
+    @transaction_completed = false
```

Comment:
I'm not sure we want to be setting this (yet) in  `#initialize_from_hash`? I don't think that `#compute_status` should ever be called on a "rehydrated" report, since it should have the status explicitly set already: https://github.com/puppetlabs/puppet/blob/274cdc39b356d3741a16ac3e0327b886beece17e/lib/puppet/transaction/report.rb#L233-L239

It's probably safe to not set that in `#initialize_from_hash` until the round-two change of (de)serializing `transaction_completed` in the report schema, and setting it from the serialized data?